### ISSUE TITLE
FMT004: Detect Markdown-style syntax in AsciiDoc files

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#30]
## Problem

LLMs trained on mixed documentation often produce Markdown syntax in AsciiDoc files:

**Common Markdown patterns in AsciiDoc (wrong):**
```asciidoc
# Heading (should be = Heading)
## Subheading (should be == Subheading)
```code``` (should be \`code\` for inline or ---- blocks)
[link text](url) (should be link:url[link text])
![alt](image.png) (should be image::image.png[alt])
**bold** (works but *bold* is more idiomatic)
> blockquote (should be [quote] block)
```

## Proposed Solution

Add a new rule `FMT004` that detects Markdown syntax patterns commonly produced by LLMs and suggests the AsciiDoc equivalent.

## Priority Patterns

1. `# ` at start of line → suggest `= `
2. `## `, `### `, etc. → suggest `== `, `=== `
3. `[text](url)` → suggest `link:url[text]`
4. `![alt](path)` → suggest `image::path[alt]`
5. Triple backticks → suggest `----` blocks

## Acceptance Criteria
- [ ] New rule `FMT004` in `format_rules.py`
- [ ] Detects common Markdown patterns
- [ ] Suggests AsciiDoc equivalents
- [ ] Tests cover various Markdown-in-AsciiDoc scenarios

---

### Architecture Context
Based on the codebase architecture (see `CLAUDE.md`):

- Core Flow: CLI → AsciiDocLinter → Parser → Rules → Finding → Reporter
- The rule will extend `Rule` base class from `asciidoc_linter/rules/base.py`
- Like `WhitespaceRule`, this rule will work with raw lines (`List[str]`) since it needs to detect Markdown patterns in the original text

### Implementation Details

1. **New File:** `asciidoc_linter/rules/format_rules.py`
   - Contains `MarkdownSyntaxRule` (FMT004)
   - Future: FMT001, FMT002, FMT003 will be added here

2. **Pattern Detection (using regex):**
   ```python
   # Markdown headings: # Heading, ## Heading, etc.
   r'^(#{1,6})\s+(.+)$'
   
   # Markdown links: [text](url)
   r'\[([^\]]+)\]\(([^)]+)\)'
   
   # Markdown images: ![alt](path)
   r'!\[([^\]]*)\]\(([^)]+)\)'
   
   # Markdown code blocks: ```
   r'^```'
   
   # Markdown blockquotes: > text
   r'^>\s+(.+)$'
   ```

3. **Registration in linter.py:**
   ```python
   from .rules.format_rules import MarkdownSyntaxRule
   # Add to rules list:
   MarkdownSyntaxRule(),
   ```

4. **Special handling in lint_string():**
   Like WhitespaceRule, this rule needs raw lines, not parsed elements.

### Test File
`tests/rules/test_format_rules.py` - Tests for all Markdown patterns
